src/couplers/base.py

```diff
@@ -0,0 +1,159 @@
+# SPDX-FileCopyrightText: 2025 EBFM Authors
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from typing import Dict
+import numpy as np
+
+from elmer.mesh import Mesh as Grid  # for now use an alias
+
+# from ebfm.geometry import Grid  # TODO: consider introducing a new data structure native to EBFM?
+from ebfm.config import CouplingConfig
+
+import logging
+from ebfm.logger import deprecation
+
+logger = logging.getLogger(__name__)
+
+
+class Coupler:
+    component_name: str  # name of this component in the coupler configuration
+
+    _couple_to_elmer_ice: bool  # whether to couple with Elmer/Ice
+    _couple_to_icon_atmo: bool  # whether to couple with ICON atmosphere
+
+    @property
+    def has_coupling(self) -> bool:
+        deprecation(
+            logger,
+            "Coupler.has_coupling is deprecated, use Coupler.has_coupling_to(comp_name) if you want to check for "
+            "specific components.\n"
+            "### Design recommendations ###\n"
+            "It is recommended to follow the strategy pattern and initialize your Coupler as a DummyCoupler, "
+            "YACCoupler or similar to avoid general has_coupling checks since this usually leads to a better design.\n"
+            "Alternatively, you can check for the type of your coupler. A DummyCoupler never has a coupling and you "
+            "can use this to perform checks similar to has_coupling.\n"
+            "##############################",
+        )
+        return_value = self._couple_to_elmer_ice or self._couple_to_icon_atmo
+        logger.debug(f"Component {self.component_name} has_coupling={return_value}")
+        return return_value
```

Comment:
I will replace `has_coupling` as suggested in this deprecation message in a follow-up PR. Similar for other cases where `deprecation` is used.